src/loaders/mongoose.js

```diff
@@ -10,25 +10,46 @@ export default async () => {
         process.exit(125);
     }
 
-    const options = {
-        user: config.db_user,
-        pass: config.db_pass,
-        dbName: config.db_name,
+    if (!(config.location_db_uri || (config.location_db_host && config.location_db_port && config.location_db_name))) {
+        console.error(`Either 'LOCATION_DB_URI' or 
+        'LOCATION_DB_HOST', 'LOCATION_DB_PORT' and 'LOCATION_DB_NAME' must be specified in the env file! 
+        See README.md for details.`);
+        process.exit(125);
+    }
+
+    const connectionOptions = {
         useNewUrlParser: true,
         useCreateIndex: true,
         useFindAndModify: false,
     };
 
+    const appDbOptions = {
+        user: config.db_user,
+        pass: config.db_pass,
+        dbName: config.db_name,
+        ...connectionOptions,
+    };
+
+    const locationDbOptions = {
+        user: config.location_db_user,
+        pass: config.location_db_pass,
+        dbName: config.location_db_name,
+        ...connectionOptions,
+    };
+
     try {
-        await mongoose.connect(config.db_uri, options);
+        await mongoose.connect(config.db_uri, appDbOptions);
+        await mongoose.createConnection(config.location_db_uri, locationDbOptions);
         await createDefaultAdmin();
     } catch (err) {
         console.error("Mongoose: failure in initial connection to the DB (aborting, will not retry)", err);
         process.exit(44);
     }
 
-    mongoose.connection.on("error", (err) => {
-        console.error("Mongoose connection error:", err);
+    mongoose.connections.forEach((connection) => {
+        connection.on("error", (err) => { // default mongoose connection used for the main app DB
```

Comment:
Is the comment accurate? Shouldn't this be iterating all the connections? Inc. the location one?

Reply:
I just copied it over, forgot to change/remove it

Reply:
Also, before this I explicitely called the on("error") on both the connection and connections[1] objects, so this is in fact iterating both connections.